bin/salt-registerd

```diff
@@ -73,106 +138,81 @@ def register_system(data):
 
             rhnSQL.commit()
 
-            logging.info("%s registered as %s" % (minion, newserv.getid()))
+            logger.info("%s registered as %s", minion, newserv.getid())
         else:
-            logging.warning("Can't get grains for %s" % minion)
-
-class Saltwalk:
-
-    def __init__(self):
-        self.event_src = salt.utils.event.MasterEvent('/var/run/salt/master')
-        self.pending_jobs = set()
-        self.local = salt.client.LocalClient()
-
-        XMLRPC_URL = "http://localhost/XMLRPC"
-        self.rpc = rpclib.Server(XMLRPC_URL)
-
-    def run(self):
-        for ev in self.event_src.iter_events(full=True):
-            # other possible values:
-            # salt/key act accept, act reject
-            # salt/auth
-            if ev['tag'] == 'salt/auth':
-                minion = ev['data']['id']
-                logging.info("minion %s authenticated" % minion)
-                # XXX do we need to do something?
-            elif ev['tag'] == 'minion_start':
-                minion = ev['data']['id']
-                logging.info('minion %s started' % minion)
-                if not self._is_registered(minion):
-                    register_system(ev)
-                self._request_package_list(minion)
-                self._request_network_ifaces(minion)
-            elif re.match(r'salt/job/\d+/new', ev['tag']):
-                # create spacewalk action if the job was not created
-                # by this reactor
-                jid = ev['data']['jid']
-                if not jid in self.pending_jobs:
-                    logging.info('Adding job %s to Spacewalk' % jid)
-                    self._create_actions_for_job(ev)
-            elif re.match(r'salt/job/\d+/ret/\S+', ev['tag']):
-                # minion return values
-                if jid in self.pending_jobs:
-                    self._process_result(jid, ev)
-                else:
-                    logging.warning('Job event %s not generated by us' % ev['tag'])
-                    try:
-                        self._update_actions_for_job(ev)
-                    except:
-                        traceback.print_exc()
-                        logging.error(ev)
-            else:
-                logging.warning('Not handling event %s' % ev['tag'])
+            logger.warning("Can't get grains for %s", minion)
+
+
+def is_minion_registered(minion):
+    """
+    Returns sid if a minion is registered, False otherwise
+    """
+    result = False
+    cursor = rhnSQL.prepare("""
+    SELECT id from rhnServer
+    WHERE digital_server_id=:did
+    """)
+    query_results = cursor.execute(did=('SALT-ID-%s' % minion))
+    if query_results > 0:
```

Comment:
Unless we are looking here for negative codes, I wouldn't do that. DBI does not defines the return codes and cx_Oracle returns `None` here, for example. Maybe it is enough to just:
```python
if cursor.execute(did=('SALT-ID-%s' % minion)):
    result = cursor.fetchone_dict().get('id', False)
return result
```